base_layer/core/src/base_node/service/service.rs

```diff
@@ -412,16 +428,30 @@ async fn handle_incoming_request<B: BlockchainBackend + 'static>(
         .handle_request(&request.try_into().map_err(BaseNodeServiceError::InvalidRequest)?)
         .await?;
 
+    // Determine if we are synced
+    let mut status_event_stream = state_machine_handle.get_status_event_stream();
+
+    let in_listening_state = match status_event_stream.recv().await {
+        None => false,
+        Some(s) => match s {
+            StatusInfo::Listening(_) => true,
+            _ => false,
+        },
+    };
+    let is_synced = in_listening_state && state_machine_handle.get_initial_blockchain_sync_success_status();
```

Comment:
As I mention later, this extra check doesn't seem necessary. Simply being in `Listening` is probably enough of a check. In fact, I'm not sure if you could ever logically have `in_listening_state == true` and `get_initial_blockchain_sync_success_status()` be false. Unless I am missing another place where is it set outside of the `Listening` state.

Reply:
Explained above why we need both these checks. After the initial sync has been completed we still need to check if we are in listening in case the node has become out of sync after the initial sync.

Reply:
Agreed with @philipr-za. When the node starts up, `in_listening_state == true` and `get_initial_blockchain_sync_success_status() == false`. Also true for a fresh node starting up for the first time.